dask_ml/model_selection/_incremental.py

```diff
@@ -849,7 +849,35 @@ def _get_params(self):
         else:
             return ParameterSampler(self.parameters, self.n_initial_parameters)
 
+    def _stop_on_plateau(self, info, instructions):
+        out = {}
+        for k, steps in instructions.items():
+            records = info[k]
+            current_calls = records[-1]["partial_fit_calls"]
+            if self.max_iter and current_calls >= self.max_iter:
+                out[k] = 0
+            elif self.patience and current_calls >= self.patience:
+                plateau = [
+                    h["score"]
+                    for h in records
+                    if current_calls - h["partial_fit_calls"] <= self.patience
+                ]
+                if all(score <= plateau[0] + self.tol for score in plateau[1:]):
+                    out[k] = 0
+                else:
+                    out[k] = steps
+
+            else:
+                out[k] = steps
+        return out
+
     def _additional_calls(self, info):
+        instructions = self._adapt(info)
+
+        out = self._stop_on_plateau(info, instructions)
+        return out
```

Comment:
I think that I would prefer to keep things as a single method until a concrete need arises to separate them.  

While I appreciate the desire to split things apart into smaller chunks I find this adds a cost of indirection, which makes code a bit harder to review and maintain in the future.  I find that code that is finely split often feels better when first writing it, but ages poorly when you have to quickly understand what is going on after a year of not looking at it. 
 There is obviously a length limit to this (functions of several hundred lines are unpleasant), but I don't think that the `_additional_calls` function is yet there.

I haven't yet reviewed the other parts of this commit that change the behavior of `_additional_calls`, partly because it's hard to separate those changes from the refactor.  If possible I'd prefer that we remove the refactor so that it's easier to review and discuss the algorithmic changes.